### DISCUSSION TITLE
希望C#可以提供方法别名 mutipart function names hope

### DISCUSSION THREAD

**Discussion by redguardofchina** [dotnet/csharplang#7045] [Language Ideas]
希望C#可以提供方法别名 mutipart function names hope

比如文本读取方法，我们可以这样定义 as a function of mutipart names

public static string GetText,GetString,ReadText,ReadString(string path);

**Comment by DavidArno** [comment]
I don't speak Mandarin, so I may have misunderstood you, but according to Google you are asking for aliases for method names, as per your code example.

Creating four methods, `GetText`, `GetString`, `ReadText`, `ReadString`, which all then do the same thing seems an unwise thing to do in my view. It creates extra cognitive load as developers, when writing and more importantly reading code using these has to take the time to check all four only to discover they do the same thing.

This feels like a “YAGNI” idea. It creates complexity through aiming to offer flexibility. Keep it simple is the key here. 

Sorry if this seems negative, but I really can’t see how this change would bring any benefits to C# developers.

**Reply by 333fred** [reply]
@DavidArno could you please rephrase your comment? Disliking an idea is fine, but the tone in this comment is a bit too much of an attack. Let me know when it's been rephrased and I'll unhide it.

**Comment by DavidArno** [comment]
Further, if you really wanted to do this, the code to do it is currently trivial and doesn't require much extra typing than your proposal:
```cs
public static string GetText(string path) => ReadString(path);
public static string GetString(string path) => ReadString(path);
public static string ReadText(string path) => ReadString(path);
public static string ReadString(string path)
{
    // code to get the string from the path
}
```

**Comment by DavidArno** [comment]


**Reply by 333fred** [reply]
Thank you David, comment is unhidden. I'll mark these as off topic now.

**Comment by huoyaoyuan** [comment]
Aliasing methods is trivial, since you can easily achieve this by defining an inlinable function calling them.
In the other hand, aliasing types are more useful. Examine a project with all identifiers from BCL being aliased to the developers' native language.
